kuzzleio/kuzzle
# Use nodemon instead of PM2 for core-dev and plugin-dev images

## What does this PR do ?

:warning: Depends on https://github.com/kuzzleio/kuzzle-containers/pull/54 :warning: 

Use Nodemon instead of PM2:
 - remove pm2 from scripts
 - remove pm2 log from core dump
 - remove pm2 from plugin-dev image
 - install nodemon in plugin-dev image
 - use nodemon in scripts
 - update [documentation](https://deploy-preview-1635--doc-kuzzle.netlify.app/guides/essentials/installing-kuzzle/)

Why ?
 - PM2 is annoying because it keep restarting when the application crash instead of waiting for changes before trying to restart
 - It has a configuration file
 - It is huge

### How should this be manually tested?

Clone the repository (you really should have a fresh start) then run `docker-compose up`

### Other changes

 - Bump Node.js version to 12.16.3
 - Deprecate AWS marketplace install since it's only Kuzzle v1
 - remove unused ` docker-compose/scripts/devtools.sh`
